apps/src/bounce/bounce.js

```diff
@@ -1003,6 +1003,7 @@ Bounce.runButtonClick = function () {
  * studioApp().displayFeedback when appropriate
  */
 var displayFeedback = function () {
+  const {isSignedIn, userType} = getStore().getState().pageConstants;
```

Comment:
I'm open to consolidating on one implementation, but:

1. I'm hoping we could move away from the patterns in progressRedux of (a) `window` globals and (b) depending on specific css classnames in the user menu, or at least not make more things depend on them

2. the logic we need here seems a bit simpler since we only care about what's in the cookie, and not whether something has changed since the page loaded (which is presumably the reason we check the user menu classnames in progressRedux, though I'm not clear on why progressRedux needs that either)

3. it seems like an odd connection in the dependency graph for individual studio apps to depend on code-studio/progressRedux (e.g. since /projects don't show progress), which seems like a worse code smell than duplicate state to me

So... I like the idea of eliminating duplicate state in general, but I'm not sure it seems like we'd come out ahead pushing for it here.

If we want to gain some yardage on consolidating state, another potential improvement would be to consolidate `pageConstants.isSignedIn` and `.userType`. I had avoided that because so many things depend on `isSignedIn` alone that I didn't want to risk breaking them if we were relying on it being false on certain HoC scripts, but it could be worth reading through and trying it if nothing looks particularly risky.

It may also be possible to pass the sign-in state down to code-studio via a more standard method such as a data attribute on a script tag and for that to be accessed by individual studio apps directly without going through progressRedux.

Reply:
1. The reason we have the window and classname stuff is because we want to be in sync with the user header JS which does not go through our pipeline. It's probably not particularly likely this changes any time soon.
2. Checking the user menu classnames isn't there to check whether something has changed since the page has loaded. It's there to make sure we're using identical logic to what populates the button in the header. I think when discussing this we'd convinced ourself the fall back would probably never actually come into play, but I wanted to match the header button logic as closely as I could.
3. I think the problem here is that some of the state in `progressRedux` really deserves its own module(s). Would this feel less smelly to you if you were just doing something like `state.signInState` instead of `state.progress.signInState`?

Reply:
Yes, state.signInState would eliminate my concern for (3). As for (1) and (2), what you've done there does accomplish the goal of making sure the logic stays in sync, but it does so in a way that seems very javascript-centric, as if to say that anything that happens in dashboard can't be trusted. I see no reason we shouldn't be able to share server logic for checking this cookie and pass it down to the client in two different places.

Reply:
The issue isn't that dashboard can't be trusted - it's that sometimes for static cached scripts, the server doesn't have any user-specific knowledge, and so cracking open the cookie is the only way we know about the user.